### DISCUSSION TITLE
What happens when myloader recovers a table with the same name?

### DISCUSSION THREAD

**Discussion by Charles7c** [mydumper/mydumper#1117] [Q&A]
What happens when myloader recovers a table with the same name? Whether you can set failure when encountering the same name or rename it to 2?

**Comment by davidducos** [comment]
Hi @Charles7c, you have this purge modes:
```
  --purge-mode                          This specify the truncate mode which can be: NONE, DROP, TRUNCATE and DELETE
```
We could add the RENAME and FAIL if the CREATE TABLE execution fails. Is that what you want? Take into account that the RENAME will be over the table in your database, which might cause other issues. I think that we should be adding FAIL and set it as default, when -o is not set, as when -o is set we should be using DROP as default. 

https://github.com/mydumper/mydumper/pull/1166

**Reply by Charles7c** [reply]
You understand me! Our requirement is like the options in rsync (https://download.samba.org/pub/rsync/rsync.1).

**Reply by Charles7c** [reply]
--backup, -b             make backups (see --suffix & --backup-dir)
--suffix=SUFFIX          backup suffix (default ~ w/o --backup-dir)
